lib/awful/client.lua

```diff
@@ -634,6 +634,42 @@ function client.object.is_fixed(c)
     return false
 end
 
+--- Is the client immobilized horizontally, i.e. fullscreen or maximized?
+--
+-- **Signal:**
+--
+--  * *property::immobilized_horizontal*
```

Comment:
Correct me if I'm wrong, but this claim of a signal is a lie?

Reply:
Isn't this for all properties?  (I assumed, but apparently there is also no test for it).

Reply:
You are right - there is no setter here.  Will remove it.

Reply:
Well, things like "fullscreen" are explicitly changed (`c.fullscreen = true`). For that case, there is some magic in place that automatically emits a signal, I think. However, this immobilized property is really a combination of other signals (fullscreen, maximized, maximized_h/v) and the magic cannot know that the getter will now suddenly return another value just because some other property changed.

Reply:
Do you want such magic? There is a few other places where it could be used (like `awful.layout.init`). Called `depends_on` or something. It would still not be able to detect if the property *didn't* change, but it would always send the signal if it *might* have changed.

Reply:
Only if we have several places that need it. So far, `immobilized_v/h` are the only two examples that I think of and right now they do not even have a signal (and I can't imagine anyone requiring it). So I'd keep things simple for now.